enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,213 @@
+
+There is no difference between MachineSet and BYOH on how we enable containerd as default runtime. 
+
+Containerd Migration plan
+   *Containerd will become default runtime as part of OpenShift 4.11
+   *For current usage, we plan to introduce WMCO feature flag and release it only for community operator.
+   *The default value will be Docker CRI. In 4.10 community branch, the default value will be containerd.
+   *Customers will not have the option to choose runtime by using WMCO feature flag.
+   *In BYOH case Windows VM Will be de-configured and re-configured with the WMCO supported runtime.
+   *In MachineSet option Docker CRI based VM will be deleted and a new VM will be created with containerd
+    as default runtime.
+
+### User Stories
+
+Stories can be found within the [Windows Containers: containerd epic](https://issues.redhat.com/browse/WINC-505)
+
+### Justification
+
+*If we don't make containerd as default runtime, we will be left out with Mirantis supported dockershim
+and will have to re-design WMCO to incorporate Mirantis dockershim.
+*From the business point of view subscribing to Mirantis for dockershim support is not an ideal situation as Mirantis
+ is our key competitor.
+*If we want to use CRI-O for Windows, the amount of time and engineering efforts involved making CRI-O to work for 
+Windows is huge. This doesn't do business justification due to the high cost and time. 
+*Containerd is widely adopted and supported by the open source community. 
+*Microsoft is a major contributor in Windows containerd development and moved to containerd as default runtime for their 
+kubernetes offerings. 
+*Most of the Windows-supported kubernetes orchestrators already moved to containerd.
+*Containerd has become the default runtime in linux/windows nodes.
+*Testing and adaptation will require less engineering effort compare to other solutions due to wider open source
+ community and enterprise support.
+
+### Design Details
+
+we plan to use containerd 1.6.0 to integrate into WMCO. Containerd will be installed as a first service before 
+kubelet installation as kubelet must require containerd to be running. We already bundled containerd as part of 
+WMCO package. As we can use ctr commandline to test containerd, crictl commandline will not be installed.
+Once containerd is installed, the rest of the service installation steps remain the same. Kubelet's dependency on Docker
+runtime will be removed and replaced by containerd. A separate folder will be created under C:\k\ to store containerd
+config and related files. In BYOH case, even after containerd becomes default runtime Docker will continue to run.
+User can choose to uninstall Docker or use it for non-kubernetes related requirements. As docker installation is
+handled by the user, it is up to the user to decide if Docker should continue to run or not.
+When it comes to MachineSet, we can choose a VM image which doesn't have Docker. Containerd will be the only runtime.
```

Comment:
We don't get to choose a VM image for the MachineSet. That is up to the cluster admin.

Reply:
I mean to say user ( cluster-admin). will correct it